app/services/update_site_setting.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class UpdateSiteSetting
+  include Service::Base
+
+  policy :current_user_is_admin
+
+  step :convert_name_to_sym
+
+  policy :setting_is_visible
+  policy :setting_is_configurable
+
+  step :cleanup_value
+  step :save
```

Comment:
A `Contract` should be defined here with `setting_name` and `new_value`  and a validation to check that at the least `setting_name` is presence (since `new_value` could be ''), this helps define the expected input for consumers.

Reply:
Good example from Kris here https://github.com/discourse/discourse/blob/e92323c81e0531797aedb1725f233b3c29c50164/app/services/reorder_flag.rb#L18-L23